src/plugins/intel_gpu/tests/functional/subgraph_tests/vlsdpa.cpp

```diff
@@ -0,0 +1,331 @@
+
+#include "openvino/opsets/opset13.hpp"
+#include "ov_ops/vl_sdpa.hpp"
+
+#include "openvino/util/log.hpp"
+#include "intel_gpu/runtime/execution_config.hpp"
+// #include "../src/graph/impls/ocl/kernel_selector_helper.h"
+
+//=================================================================================
+// Transpose + SDPA + Transpose pattern fusion (TransposeVLSDPAMatcher)
+// In this test case, SDPA is first transformed to VLSDPA under desired situations.
+// Then the pattern is transformed to VLSDPA after pass of TransposeVLSDPAMatcher
+// is applied.
+//=================================================================================
+namespace ov {
+namespace test {
+using namespace ov;
+using namespace ov::opset13;
+using namespace ov::intel_gpu;
+
+using TransposeVLSDPATestParams = std::tuple<ElementType,
+                                             std::vector<int32_t>>;  // cu_seqlens
+
+class TransposeVLSDPATestOnGPU: public testing::WithParamInterface<TransposeVLSDPATestParams>,
+                                virtual public test::SubgraphBaseTest {
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<TransposeVLSDPATestParams> obj) {
+        ElementType inType;
+        std::vector<int32_t> cu_seqlens;
+
+        std::tie(inType, cu_seqlens) = obj.param;
+
+        std::ostringstream result;
+        result << "device=(" << std::string(utils::DEVICE_GPU) << ")_";
+        result << test::utils::vec2str<int32_t>({cu_seqlens}) << "_";
+        result << "Prc=" << inType;
+        return result.str();
+    }
+
+    static std::shared_ptr<Parameter> make_param(const PartialShape& pshape,
+                                                element::Type element_type,
+                                                const std::string& name) {
+        auto param = std::make_shared<Parameter>(element_type, pshape);
+        param->set_friendly_name(name);
+        param->get_output_tensor(0).set_names({name});
+        return param;
+    }
+
+    static bool check_vlsdpa_available() {
+        ElementType inType;
+        std::vector<int32_t> cu_seqlens;
+        std::tie(inType, cu_seqlens) = GetParam();
+        if (inType != ElementType::f16) // VLSDPA CM kernel supports half precision only
+            return false;
+        // VLSDPA CM optimized for Xe1/Xe2 architectures only
+        // auto engine = cldnn::engine::create(cldnn::engine_types::ocl, cldnn::runtime_types::ocl);
+        // const auto& info = engine->get_device_info();
+        // if (!(info.arch >= cldnn::gpu_arch::xe_lp)) {
+        //     return false;
+        // }
+
+        // ExecutionConfig config;
+        // if (!check_cm_jit_support(*engine.get(), config)) {
+        //     OPENVINO_WARN("You may miss SDPAToVLSDPA optimization for QWenVL model,"
+        //                 "as CM environment is unavailable. Enable it by installing proper GPU driver and CM compiler.");
+        //     return false;
+        // }
```

Comment:
@sshlyapn  This unit test should only execute on machines with desired IGC version installed. It looks "check_cm_jit_support(*engine.get(), config)" cannot work here, as it lacks "engine" and "config" info. And the function declaration is not visible to functional test. I have to move it to some utility header. What would you suggest?